File: 2022-2023/sourcecode/lab/lab13/DaggerHiltROOMBasicsJava/app/src/main/java/pl/edu/uwr/pum/dagger_hilt_room_basicsjava/data/repository/AppRepositoryImpl.java
package pl.edu.uwr.pum.dagger_hilt_room_basicsjava.data.repository;

import androidx.lifecycle.LiveData;

import java.util.List;

import javax.inject.Inject;

import pl.edu.uwr.pum.dagger_hilt_room_basicsjava.data.Student;
import pl.edu.uwr.pum.dagger_hilt_room_basicsjava.data.db.AppDao;
import pl.edu.uwr.pum.dagger_hilt_room_basicsjava.domain.repository.AppRepository;

public class AppRepositoryImpl implements AppRepository {

    private final AppDao appDao;

    @Inject
    public AppRepositoryImpl(AppDao appDao) {
        this.appDao = appDao;
    }

    @Override
    public LiveData<List<Student>> readAllData() {
        return appDao.readAllData();
    }

    @Override
    public void insert(Student student) {
        appDao.addItem(student);
    }
}

File: 2022-2023/sourcecode/lab/lab13/DaggerHiltROOMBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/dagger_hilt_room_basicskotlin/data/repository/AppRepositoryImpl.kt
package pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.data.repository

import androidx.lifecycle.LiveData
import pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.data.Student
import pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.data.db.AppDao
import pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.domain.repository.AppRepository
import javax.inject.Inject

class AppRepositoryImpl @Inject constructor (private val appDao: AppDao) : AppRepository {
    override val readAllData: LiveData<List<Student>> = appDao.readAllData()
    override suspend fun insert(student: Student) = appDao.addItem(student)
}